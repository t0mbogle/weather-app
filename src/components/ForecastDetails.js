import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forcast-details">
      <div className="forecast-details_date">{formattedDate}</div>
      <div className="forecast-details_temp">{temperature.min}˚C</div>
      <div className="forecast-details_temp">{temperature.max}˚C</div>
      <div className="forecast-details_humidity">{humidity}</div>
      <div className="forecast-details_wind">
        {wind.speed}
        {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape(
    PropTypes.shape({
      date: PropTypes.string,
      humidity: PropTypes.number,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
      }),
      wind: PropTypes.shape({
        speed: PropTypes.number,
        direction: PropTypes.string,
      }),
    })
  ).isRequired,
};
