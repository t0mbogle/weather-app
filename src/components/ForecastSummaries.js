import PropTypes from "prop-types";
import React from "react";
import ForecastSummary from "./ForecastSummary";

// .map transforms each forecast into a ForecastSummary component
function ForecastSummaries({ forecasts }) {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          // key is set to the forecast.date as this is should be unique to each instance of forecast
          key={forecast.date}
          date={forecast.date}
          description={forecast.description}
          icon={forecast.icon}
          temperature={forecast.temperature}
        />
      ))}
    </div>
  );
}

export default ForecastSummaries;

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
      }),
    })
  ).isRequired,
};
