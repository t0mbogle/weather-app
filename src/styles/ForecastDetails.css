.forcast-details {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
}