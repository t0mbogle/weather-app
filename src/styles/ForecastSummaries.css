.forecast-summaries {
    display: flex;
    justify-content: space-between;
    margin-top: 40px;
    margin-bottom: 5%;
}

