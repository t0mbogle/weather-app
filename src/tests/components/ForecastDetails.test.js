import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 80,
    temperature: {
      min: 8,
      max: 13,
    },
    wind: {
      speed: 60,
      direction: "ne",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details_date");
    expect(getByText(/80/)).toHaveClass("forecast-details_humidity");
    expect(getByText("13ËšC")).toHaveClass("forecast-details_temp");
    expect(getByText(/60/)).toHaveClass("forecast-details_wind");
  });
});
