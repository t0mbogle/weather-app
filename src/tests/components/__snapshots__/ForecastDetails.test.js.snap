// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forcast-details"
  >
    <div
      class="forecast-details_date"
    >
      <b>
        Date:
      </b>
       Thu Jan 01 1970
    </div>
    <div
      class="forecast-details_temp"
    >
      <b>
        Max Temperature:
      </b>
       13˚C
    </div>
    <div
      class="forecast-details_temp"
    >
      <b>
        Min Temperature:
      </b>
       8˚C
    </div>
    <div
      class="forecast-details_humidity"
    >
      <b>
        Humidity:
      </b>
       80%
    </div>
    <div
      class="forecast-details_wind"
    >
      <b>
        Wind: 
      </b>
      60ne
    </div>
  </div>
</DocumentFragment>
`;
